pkg/controllers/externalsecret/externalsecret_controller.go

```diff
@@ -320,36 +342,47 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	}, nil
 }
 
-func createOrUpdate(ctx context.Context, c client.Client, obj client.Object, f func() error, fieldOwner string) error {
+func isOrphanSecret(externalSecret esv1beta1.ExternalSecret, newSecret *v1.Secret) bool {
+	secretRef := externalSecret.Status.CreatedSecretReference
+	if externalSecret.Spec.Target.CreationPolicy != esv1beta1.CreatePolicyOwner ||
+		externalSecret.Status.CreatedSecretReference == nil ||
```

Comment:
nit: ...because it has been defined above, we should use it for brevity

```suggestion
		secretRef == nil ||
```